articles/introduction-to- virtual- environments-and-dependency-managers/index.md

```diff
@@ -0,0 +1,419 @@
+﻿### Introduction to Python Virtual Environments and Dependency managers
+
+Virtual environments. What are they and what are they used for? From the Python documentation, a virtual environment is an isolated Python environment where a Python interpreter, libraries, modules and scripts (collectively called dependencies) are installed in a different directory from those installed in the system’s default Python path and other virtual environments.
+
+Dependency managers are tools that enable easy management of a project's dependencies.
+
+### Prerequisites
+
+This article assumes:
+
+* The reader's knowledge of Python programming language.
+* Has Python installed into his/her computer.
+
+    >Just to make sure run this command on your terminal.  `python --version`
+
+### Overview
+
+This article will look into the most used virtual environments and dependency managers in python, how to use them and their pros and cons.
+
+1. [Pipenv](#1.-Pipenv)
+2. [Venv](#2.-Venv)
+3. [Virtualenv](#3.-Virtualenv)
+4. [Poetry](#4.-Poetry)
+5. [Conda](#5.-Conda)
+
+### 1.  Pipenv
+
+### What is Pipenv?
+
+Pipenv is a package manager that is used to manage python projects dependencies. It is good for the common typical workflow of using dependencies like *pip*, *requirements.txt* etc.
+Under the hood, pipenv generally combines the use of pip and Virtualenv into a simple command line interface.
+
+### The problem it solves
+
+1. Managing projects with different dependencies i.e. when one project requires `Django==2.6` but another project requires `Django==3.0.6`.
+2. Installation of python packages and libraries.
+3. Dependency resolution. This is where you specify the requirements for a particular project sub-dependency to avoid installation problems.
+4. Reproduction of environments
+
```

Comment:
These problems are solved by all the virtual environments managers. Please consider moving them to a general section.